File: MedicalAppointments_UPDATED/project/ProjectFinal/backend/MedicalAppointments.API/Models/WaitingListEntry.cs
namespace MedicalAppointments.API.Models;

public class WaitingListEntry
{
    public int Id { get; set; }
    public int DoctorId { get; set; }
    public DateOnly WeekStartDate { get; set; }
    public string PatientName { get; set; } = default!;
    public string Email { get; set; } = default!;
    public string? Phone { get; set; }
    public DateTime? NotifiedAt { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    // Navigation
    public User Doctor { get; set; } = default!;
}

File: appointment-service/src/main/java/com/hospital/appointment/entity/WaitingListEntry.java
package com.hospital.appointment.entity;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;
import java.time.LocalDateTime;

/**
 * Inscription sur la liste d'attente d'un médecin pour une semaine donnée.
 * Migré depuis Models/WaitingListEntry.cs (.NET).
 *
 * Note : la table est nommée "WaitingList" (singulier) comme dans .NET.
 */
@Entity
@Table(name = "WaitingList", uniqueConstraints = {
        @UniqueConstraint(
                name = "ux_waitinglist_doctor_week_email",
                columnNames = {"doctorId", "weekStartDate", "email"})
})
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class WaitingListEntry {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(name = "doctorId", nullable = false)
    private Integer doctorId;

    @Column(nullable = false)
    private LocalDate weekStartDate;

    @Column(nullable = false)
    private String patientName;

    @Column(nullable = false)
    private String email;

    private String phone;

    private LocalDateTime notifiedAt;

    @Column(nullable = false)
    @Builder.Default
    private LocalDateTime createdAt = LocalDateTime.now();

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "doctorId", referencedColumnName = "id",
            insertable = false, updatable = false)
    @ToString.Exclude
    @EqualsAndHashCode.Exclude
    private User doctor;

    @PrePersist
    public void prePersist() {
        if (createdAt == null) createdAt = LocalDateTime.now();
    }
}
